### ISSUE TITLE
CMake: Potential Out of Order building causing errors depending on Generator/OS/etc

### ISSUE THREAD

**Issue by capnkenny** [novelrt/NovelRT#481]
Self-titled - essentially, depending on a number of different factors (which are hard to isolate), there is a potential for CMake to try and build things out of order even though there are dependencies listed, causing numerous errors, such as:

- 0xc0000135 on Windows (due to not copying the DLLs at the right time)
- Linking errors (missing linker files due to the dependencies themselves not being linked yet)

It seems that the more threads a device has to build with, the less likely this is actually going to happen - it's _prominent_ to happen on CI for Windows 2019/2022, but locally and in Ubuntu/macOS it's not reproducible at this time. We may want to revamp our CMakeLists soon to get everything in order and to stop this from occurring.

**Comment by RubyNova** [comment]
Is this CI only? Is it going to affect our release?

**Comment by RubyNova** [comment]
What happens if we force single-threaded?

**Comment by RubyNova** [comment]
I'm just worried this is going to suddenly break on us. It shouldn't be doing this.

**Comment by capnkenny** [comment]
I don't think you have to worry that this should suddenly break - however I will say that if we continue to further add dependencies the way we are currently doing so, it's going to become more difficult to manage and this _may_ show itself more without additional changes.
